js/Listeners.js

```diff
@@ -136,9 +136,12 @@ function Listeners() {
  *
  * @param {String} event Event name
  * @param {Function} callback Function to call when an even of the type is received
- * @return {int} The index of the new listener
+ * @return {int|undefined} The index of the new listener
  */
 Listeners.prototype.add = function (event, callback) {
+	if (typeof callback !== 'function') {
```

Comment:
This is a new. I don't think there's any reason to accept callbacks which are not functions. Shoud maybe throw an error instead?